filament/backend/src/metal/MetalHandles.h

```diff
@@ -187,6 +196,17 @@ class MetalFence : public HwFence {
     uint64_t value;
 };
 
+struct MetalTimerQuery : public HwTimerQuery {
+    MetalTimerQuery() : status(std::make_shared<Status>()) {}
+
+    struct Status {
+        std::atomic<uint64_t> elapsed;
```

Comment:
I'm pretty sure elapsed doesn't need to be atomic because it's protected by `available`. In other words, if someone reads it before `available` is `true`, it doesn't matter if it's garbage. On the other hand if `available` is observed as true through a "load acquire", it means you have done an atomic "store release", so `elapsed` should be seen as written.